docs/content/docs/deep/componets.md

```diff
@@ -0,0 +1,113 @@
+Some exceptions to this are known to work: TODO: list working/supported exceptions here.
+
+An application can select which DDses (TODO: or DataObjects?) in the container to load, allowing both reading and writing to containers which are only partially loaded.
+(TODO: note scalability implications due to summary client, op bandwidth).
+
+Within a container, all DDses are synchronized, and operations can between different DDSes are ordered.
+Each DDS has its own merge resolution, so there is no way to do things like multi DDS transactions, or have changes in one DDS impact how a change is merged in another, which can limit the fidelity of operations like moves of content between DDSes, even within a container.
+
+The Fluid Framework provides a garbage collector which can delete DDSes when they become unreferenced by the container (no reachable from the root DDS, which is typically the shared directory created from the container schema).
+
+### Data Objects
+
+Data objects are an under development abstraction on-top of DDSes (see below) which can allow reusable logic build on DDSes to be easily reused.
+
+Currently no public APIs exist for defining new data-object types, but the existing experimental ones can be used in container schema.
+
+### Distributed Data Structures: DDSes
+
+A data structure in a container, with its own self contained editing and merge resolution logic.
+May contain handles to other DDSes.
+
+Some DDSes (mainly SHaredTree) provide their own functionality for further componentization of their content.
+
+While while the user of the container can choose which DDSes to load, DDSes themselves currently can not support being only partially loaded: either the whole DDS is available or not.
+The one exception to this is blobs which the user of the container can asynchronously upload then insert handle to them into the DDS.
+Future work (See SharedTree) may enable some DDSes to load subsets of their data on demand.
+
+## SharedTree
+
+SharedTree is design to encourage componentization by subtree / schema.
+Typical usage is a given component defines its schema and logic which works with that schema.
+This schema/component is then referenced/depended on by its parent, adding its schema as a child.
+
+If an application is using a design pattern with a separate Model and View,
+it should be practical to either package the view code for the component alongside its model or separately.
+
+If the same data (same schema) needs to be handled differently in different parts of the application, that logic (anything which differs) should be separate from the schema.
+Anything which is always the same however can be included as methods or additional state declared as part of the schema class if this is helpful.
+While the entire tre is branched, creating a branch can be done from just a subtree, so as long as the rest of the tree isn't modified, its indistinguishable from just branching the specific subtree.
```

Comment:
I'm not sure what exactly you are trying to say here. That you do not have to have access to the root to be successful?